A38-admin-interface-api.md

```diff
@@ -0,0 +1,200 @@
+* [Client Status Discovery Service](https://github.com/envoyproxy/envoy/blob/main/api/envoy/service/status/v3/csds.proto): the xDS config dump library that dumps the accepted xDS configuration with config status;
+
+However, the number of admin services might increase in the future. Solutions with no code changes, like env, or config files, are not desired (see rationale section below). On the other hand, as a library, gRPC would like to promote the usage of admin services. So, there is a need for an ideally one LOC to create and start an admin server.
+
+In addition, by providing new APIs, we abstract away the individual services with the concept admin services. We can also achieve **dynamic admin services selection**, for example, we **only add CSDS service if the xDS package is pulled in**. In this way, we won’t force users to take unnecessary dependencies.
+
+
+### Types of Services
+
+To help communicate the scenarios, we can define all services into several types, from the view of administrative functions:
+
+* **Primary service**: the user-provided services that handle the application logic;
+* **Binary-global service**: the services that expose the state of the entire gRPC library, and they serve the same information regardless of which server they have been registered (e.g., Channelz, CSDS);
+* **Per-server service**: the services that distribute the state of other registered services on the same server (e.g., Health, Reflection).
+
+The binary-global services contain sensitive information about traffic, configuration, even security, hence we hope users can set up the services securely in our recommended usage or the upcoming new API.
+
+The recommended way of securing the admin service port is binding it to either `localhost` or using a `UDS` port.
+
+
+## Proposal
+
+There are several different options for the convenience API design to help create a more usable admin interface API. Each option is a tradeoff between simplicity to use and flexibility. The simplicity to use is defined by lines of code to create a serving gRPC server with admin services, and the additional cognitive cost to learn the new API or manage dependencies.
+
+The flexibility is defined by how can users provide additional server creation options. To date, users can specify one or more **listening addresses, credentials, TCP/HTTP config, codec, interceptors and service implementations** during the server creation process. Not to mention the potential  impact on the threading model during the start of the server. If we want to bundle as many options as possible into one line of code, we will inevitably sacrifice some flexibility.
+
+Options to create an admin server are ordered from the most flexible to the least:
+
+1. The control group: using existing API to create the server;
```

Comment:
This section should describe only the option that we're actually going with.  Discussion of the remaining approaches that we considered should be moved down to the "Rationale" section.

Reply:
Updated. This section now only talks about the picked option, and options are no longer numbered, since they will be discussed in place.